src/simplify_algebra.cpp

```diff
@@ -928,21 +928,26 @@ struct find_conv_dot_horiz_fusion
             int64_t offset = 0;
             for(auto arg : range(start, last))
             {
-                auto outputs = arg->outputs();
-                for(auto output : outputs)
-                {
-                    if(output->name() != "reshape")
-                        continue;
-                    auto x = m.insert_instruction(output, make_op("contiguous"), arg);
-                    m.replace_instruction(output, output->get_operator(), x);
-                }
+                auto outputs             = arg->outputs();
+                auto requires_contiguous = std::any_of(outputs.begin(), outputs.end(), [](auto o) {
+                    return o->get_shape().standard();
+                });
```

Comment:
This doesn't look to be a good check since the output might be a standard shape even if the next instruction does not need a standard shape. We should instead check if the next instruction requires a standard shape. We probably need a new matcher that checks the operation `attributes()` for `"require_std_shape"`. I'm not too familiar with the matcher syntax though.

Reply:
Actually, I believe you can just get the next instruction, convert it to a value, and then check the `attributes`.

Reply:
Ya I agree, the idea behind the loose check was that the eliminate_contiguous pass will always take care of the unnecessary contiguous ops added. Let me see if doing it based on that attribute works, that sounds better

Reply:
Checking for the "require_std_shape" attribute fails for the new test case I added